### PR TITLE
Device credentials endpoint

### PR DESCRIPTION
**Pull request by ignaciojonas**

### CODE REVIEW

File: lib/auth0/api/v2/device_credentials.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          raise Auth0::InvalidParameter, 'Must supply a valid device_id' if device_id.to_s.empty?
+          raise Auth0::InvalidParameter, 'Must supply a valid client_id' if client_id.to_s.empty?
+          request_params = {
+            device_name:  device_name,
+            type:         type,
+            value:        value,
+            device_id:    device_id,
+            client_id:    client_id
+          }
+          # authorization: "Basic base64(\"Username-Password-Authentication\\test:test\")"
```
Comment (ignaciojonas) [inline lib/auth0/api/v2/device_credentials.rb]: This comment should be removed.

File: spec/integration/lib/auth0/api/v2/api_device_credentials_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      "#{user['name']}_phone_1",
+      'dmFsdWU=',
+      '68753A44-4D6F-1226-9C60-0050E4C00067',
+      ENV['CLIENT_ID']
+    )
+  end
+
+  after(:all) do
+    client.delete_user(user['user_id'])
+    credentials = basic_client.device_credentials(ENV['CLIENT_ID'])
```
Comment (ignaciojonas) [inline spec/integration/lib/auth0/api/v2/api_device_credentials_spec.rb]: We should check if deleting the user also deletes the device credentials

File: lib/auth0/api/v2/device_credentials.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            raise Auth0::InvalidParameter, 'Type must be one of \'public_key\', \'refresh_token\''
+          end
+          get(device_credentials_path, request_params)
+        end
+        alias list_device_credentials device_credentials
+
+        # Creates a new device public key.
+        # @see https://auth0.com/docs/api/v2#!/device_credentials/post_device_credentials
+        # @param device_name [string] The device's name, a value that must be easily recognized by the device's owner.
+        # TODO: test this affirmation. If true, add constraint. Else, remove 'must be' from comment and default from code
```
Comment (ignaciojonas) [inline lib/auth0/api/v2/device_credentials.rb]: We should complete these TODO

File: spec/lib/auth0/api/v2/device_credentials_spec.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    it do
+      expect { @instance.create_device_credential('device_name', 'value', nil, 'client_id', 'type') }
+        .to raise_error('Must supply a valid device_id')
+    end
+    it do
+      expect { @instance.create_device_credential('device_name', 'value', 'device_id', nil, 'type') }
+        .to raise_error('Must supply a valid client_id')
+    end
+  end
+
```
Comment (falvarisqueta) [inline spec/lib/auth0/api/v2/device_credentials_spec.rb:64]: Might be good to test exception when credentials type is different than 'public_key' or 'refresh_token'

Comment (fleko) [inline spec/lib/auth0/api/v2/device_credentials_spec.rb:64]: @falvarisqueta: will remove the type parameter from this method. It must forcefully be "public_key" on creation.

File: lib/auth0/mixins/initializer.rb [hunk 5]
```diff
@@ -25,12 +25,23 @@ def authorization_header(token)
...
+        self.class.headers 'Authorization' => "Basic #{Base64.strict_encode64("#{connection_id}\\#{user}:#{password}")}"
+      end
+
       private
 
       def initialize_api(options)
         api_v1?(options) ? initialize_v1(options) : initialize_v2(options)
         raise InvalidCredentials, 'Must supply a valid API token' if @token.nil?
-        authorization_header(@token)
+        if options.fetch(:authorization, nil) == 'Basic'
```
Comment (ignaciojonas) [inline lib/auth0/mixins/initializer.rb:40]: @leosoubeste This will impact in the migration to rest-client. Just to let you know. Not action needed now.

**Comment by ignaciojonas** [comment]
Looks good for me. @falvarisqueta what do you think?

**Comment by falvarisqueta** [comment]
Looks good to me

**Pull request merged**